lib/ts/recipe/dashboard/api/user/create/emailpasswordUser.ts

```diff
@@ -58,25 +66,49 @@ export const createEmailPasswordUser = async (
         });
     }
 
-    const validateEmailError = await defaultEmailValidator(email);
+    const emailFormField = emailPasswordOrThirdpartyEmailPassword.config.signUpFeature.formFields.find(
+        (field) => field.id === "email"
+    );
 
-    if (validateEmailError !== undefined) {
-        return {
-            status: "INPUT_VALIDATION_ERROR",
-            message: validateEmailError,
-        };
+    if (emailFormField !== undefined) {
+        const validateEmailError = await emailFormField.validate(email, tenantId, userContext);
+
+        if (validateEmailError !== undefined) {
+            return {
+                status: "EMAIL_VALIDATION_ERROR",
+                message: validateEmailError,
+            };
+        }
+    } else {
+        // this should never happen.
+        throw new Error("emailFormFiled is undefined");
     }
```

Comment:
so why is this block there? Just do `emailFormField!. validate(...)`

Reply:
done